server/plugin/metrics.go

```diff
@@ -0,0 +1,11 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package plugin
+
+type metricsInterface interface {
```

Comment:
Some mild concerns here. This is duplicating the interface, which means that if one has to add a method to this, it needs to be added in 2 places.

Reply:
If we weren't constrained by the enterprise repository, would we still be using the monolithic `einterfaces.MetricsInterface`? I'm wondering if smaller, local  interfaces that reduce coupling is where we'd want to end up anyway.

In any case, I suspect this is a "temporary" change until we can refactor the parts of the `plugin` package that have to do with running the plugin sever side out of `public` altogether. None of this has to be exposed to clients.

Reply:
Sounds good, I'll plan to track this among other `public/` clean up items once we get it all live.